### ISSUE TITLE
What is the use for `cls_threshold_to_reg` in Spars4DHead?

### ISSUE THREAD

**Issue by jasonkhadka** [linxuewu/Sparse4D#24]
Using this while training seems to just drop all of gts and predictions with mask being basically 0s. 

This parameter is defined as `0.05` in configs:

https://github.com/linxuewu/Sparse4D/blob/db55d91b0b841af46d1a5dbf2393c7a26ac3f7e3/projects/configs/sparse4d_r101_H1.py#L49

But while running the training, this line causes problem:
https://github.com/linxuewu/Sparse4D/blob/db55d91b0b841af46d1a5dbf2393c7a26ac3f7e3/projects/mmdet3d_plugin/models/sparse4d_head.py#L245-L247

Basically,  all of the `cls` scores are below the threshold and thus mask's elements are set to 0s by this `torch.logical_and` gate.

**Comment by linxuewu** [comment]
Please check the version of mmdet3d wheather >= 1.0.0rc0.

> Using this while training seems to just drop all of gts and predictions with mask being basically 0s.
> 
> This parameter is defined as `0.05` in configs:
> 
> https://github.com/linxuewu/Sparse4D/blob/db55d91b0b841af46d1a5dbf2393c7a26ac3f7e3/projects/configs/sparse4d_r101_H1.py#L49
> 
> But while running the training, this line causes problem:
> 
> https://github.com/linxuewu/Sparse4D/blob/db55d91b0b841af46d1a5dbf2393c7a26ac3f7e3/projects/mmdet3d_plugin/models/sparse4d_head.py#L245-L247
> 
> Basically, all of the `cls` scores are below the threshold and thus mask's elements are set to 0s by this `torch.logical_and` gate.

**Comment by jasonkhadka** [comment]
What difference does it make to this parameter with mmdet3d version?

> Please check the version of mmdet3d whether >= 1.0.0rc0.

**Comment by linxuewu** [comment]
> What difference does it make to this parameter with mmdet3d version?
> 
> > Please check the version of mmdet3d whether >= 1.0.0rc0.

The definition of yaw before version 1.0.0rc0 is different from the definition in our code, which causes the training model to converge to an abnormal local optimum, and all cls tend to be close to 0.
